opentelemetry-resourcedetector-gcp/src/opentelemetry/resourcedetector/gcp_resource_detector/_gce.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import re
+from dataclasses import dataclass
+
+from opentelemetry.resourcedetector.gcp_resource_detector._metadata import (
+    MetadataAccessException,
+    get_metadata,
+)
+
+# Format described in
+# https://cloud.google.com/compute/docs/metadata/default-metadata-values#vm_instance_metadata
+_ZONE_REGION_RE = re.compile(
+    r"projects\/\d+\/zones\/(?P<zone>(?P<region>\w+-\w+)-\w+)"
+)
+
+_logger = logging.getLogger(__name__)
+
+
+def on_gce() -> bool:
+    try:
+        get_metadata()["instance"]["machineType"]
+    except (MetadataAccessException, KeyError):
+        _logger.debug(
+            "Could not fetch metadata attribute instance/machineType, "
+            "assuming not on GCE.",
+            exc_info=True,
+        )
+        return False
+    return True
+
+
+def host_type() -> str:
+    return get_metadata()["instance"]["machineType"]
+
+
+def host_id() -> str:
+    return get_metadata()["instance"]["id"]
+
+
+def host_name() -> str:
+    return get_metadata()["instance"]["hostname"]
```

Comment:
I think we want instance/name instead of instance/hostname

Reply:
Ah nice catch. I'll have to fix in JS as well https://github.com/GoogleCloudPlatform/opentelemetry-operations-js/blob/786cec33b16a0b8c776daa00c6771320bc217b21/packages/opentelemetry-resource-util/src/detector/gce.ts#L66